### PR TITLE
Refactor wallet private key config

### PR DESCRIPTION
**Pull request by huck-hgraph**
### Summary
- Renames env var to `NEXT_PUBLIC_HEDERA_PRIVATE_KEY`
- Refactors `HederaWallet` to use static init method
- Add `key` prop to mapped components

Note: Will add similar config in the DApp in a following PR

### CODE REVIEW

File: wallets/react-wallet-v2/.env.local.example [hunk 1]
```diff
@@ -1,2 +1,7 @@
 NEXT_PUBLIC_PROJECT_ID=
 NEXT_PUBLIC_RELAY_URL=wss://relay.walletconnect.com
+NEXT_PUBLIC_HEDERA_ACCOUNT_ID=
+# WARNING: Do not use this approach of storing/accessing private keys in production.
+# This app is not configured with a secure storage mechanism, so this is just the
+# best way to make it work and showcase Hedera integration with WalletConnect.
+NEXT_PUBLIC_HEDERA_PRIVATE_KEY=
```
Comment (hgraphql) [inline wallets/react-wallet-v2/.env.local.example:7]:
```suggestion
# !!! WARNING!!!: 
# Do not use this approach of storing/accessing private keys in production.
# This is for development purposes only, this key will be publicly available in an untrusted environment - the browser
# As this demo app is not configured with a secure storage mechanism, this method of adding a Hedera Testnet account is for demonstration purposes only!
# To create a Hedera Testnet account, proceed to https://portal.hedera.com
NEXT_PUBLIC_HEDERA_PRIVATE_KEY=
```

Comment (hgraphql) [reply wallets/react-wallet-v2/.env.local.example:7]: added exclamation points to really call it out. Also, put where they can create a testnet account in another spot, and after reading fully...it's well called out in the README.md ..though still think it's good here

File: wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts [hunk 2]
```diff
@@ -5,17 +5,23 @@ export let hederaWallet: HederaWallet
...
 export async function createOrRestoreHederaWallet() {
   try {
-    const accountId = process.env.NEXT_PUBLIC_HEDERA_ACCOUNT_ID
-    const privateKey = process.env.HEDERA_PRIVATE_KEY
-    if (!accountId || !privateKey) {
+    /**
+     * WARNING: Do not use this approach of storing/accessing private keys in production.
+     * This app is not configured with a secure storage mechanism, so this is just the
+     * best way to make it work and showcase Hedera integration with WalletConnect.
+     */
```
Comment (hgraphql) [inline wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts:12]:
```suggestion
   /**
     * !!!WARNING!!!
     *  Do not use this approach of storing/accessing private keys in production.
     * This app is not configured with a secure storage mechanism.
     * This method of key storage is only meant to showcase Hedera integration with WalletConnect and is not appropriate for sensitive keys.
     */
```

**Review by hgraphql — APPROVED** [review]
added suggestions for some wording, feel free to alter. looks great

File: wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts (hunk 2 above)
Comment (huck-hgraph) [reply wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts:12]: Nice. I'll do that

**Pull request merged**